pax-web-jetty/src/main/java/org/ops4j/pax/web/service/jetty/internal/JettyServerWrapper.java

```diff
@@ -244,28 +232,18 @@ void removeContext(final HttpContext httpContext) {
 			if (context == null) {
 				return;
 			}
-			int nref = context.decrementRefCount();
-			if (nref <= 0) {
-				try {
-					readLock.unlock();
-					writeLock.lock();
-					LOG.debug(
-							"Removing ServletContextHandler for HTTP context [{}].",
-							httpContext);
-					context = contexts.remove(httpContext);
-				} finally {
-					readLock.lock();
-					writeLock.unlock();
-				}
-			} else {
-				LOG.debug(
-						"ServletContextHandler for HTTP context [{}] referenced [{}] times.",
-						httpContext, nref);
-				return;
-			}
-		} finally {
-			readLock.unlock();
-		}
+                } finally {
+                        readLock.unlock();
+                }
+			
+                try {
+                        writeLock.lock();
+                        LOG.debug("Removing ServletContextHandler for HTTP context [{}].", httpContext);
+                        context = contexts.remove(httpContext);
+                } finally {
+                        writeLock.unlock();
+                }
+                
 		// Destroy the context outside of the locking region
```

Comment:
It's related to removing reference counting - `context.decrementRefCount()` it's called only once. So this method was no-op, as `if (nref <= 0)` is always false. Context class keeps permanent (not removable by public API) reference to listeners added in web.xml, via `_durableListeners` field.

I'm not sure why this ref counting was introduced - to prevent accidental removals or to lock item during multithreading (I've got patch with releasing ref counting, too).